modules/openapi-generator/src/main/resources/typescript/configuration.mustache

```diff
@@ -105,43 +105,36 @@ export function createConfiguration(conf: ConfigurationParameters = {}): Configu
  */
 {{#useInversify}}
 export function mergeConfiguration(conf: Configuration, options?: Configuration): Configuration {
-    if (options) {
-        conf = options;
-    }
-    return conf;
+    return options || conf;
 }
 {{/useInversify}}
 {{^useInversify}}
 export function mergeConfiguration(conf: Configuration, options?: ConfigurationOptions): Configuration {
-    let allMiddleware: Middleware[] = [];
-    if (options && options.middleware) {
-        const middlewareMergeStrategy = options.middlewareMergeStrategy || "replace" // default to replace behavior
-        // call-time middleware provided
-        const calltimeMiddleware: Middleware[] = options.middleware;
+    if (!options) {
+        return conf;
+    }
+    return {
+        baseServer: options.baseServer || conf.baseServer,
+        httpApi: options.httpApi || conf.httpApi,
+        authMethods: options.authMethods || conf.authMethods,
+        middleware: mergeMiddleware(conf.middleware, options?.middleware, options?.middlewareMergeStrategy),
+    };
+}
 
-        switch(middlewareMergeStrategy) {
+function mergeMiddleware(staticMiddleware: Middleware[], calltimeMiddleware?: Middleware[], strategy?: "append" | "prepend" | "replace") {
+    if (!calltimeMiddleware) {
+        return staticMiddleware;
+    }
+    // default to replace behavior
+    switch(strategy ?? "replace") {
```

Comment:
can this become a signature change, i.e.: `..., strategy: "append" | "prepend" | "replace" = "append"` then the default is also visible for callers.

Reply:
Good idea. Changed :-)